### PR TITLE
Add bundling shared libs with jars

### PR DESCRIPTION
**Pull request by Solpatium**

### CODE REVIEW

File: lib/src/main/kotlin/starknet/crypto/NativeLoader.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            // Find the package bundled in this jar
+            val resource = sharedLibExtensions.map {
+                val fileName = "/lib$name.$it"
+                object {}::class.java.getResource(fileName)
+            }.firstOrNull { it != null } ?: throw e
+
+            val tmpDir = Files.createTempDirectory("$name-dir").toFile().apply {
+                deleteOnExit()
+            }
+            val tmpFile = File(tmpDir, name)
```
Comment (cptartur) [inline lib/src/main/kotlin/starknet/crypto/NativeLoader.kt:27]: I think it doesn't actually create a file in the filesystem, just the java object.

It should be replaced with `Files.createFile()`

File: lib/src/main/kotlin/starknet/crypto/NativeLoader.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            resource.openStream().use { Files.copy(it, tmpFile.toPath()) }
```
Comment (cptartur) [inline lib/src/main/kotlin/starknet/crypto/NativeLoader.kt:28]: I think `Files.copy` would be simpler.

File: lib/src/main/kotlin/starknet/crypto/NativeLoader.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package starknet.crypto
+
+import java.io.File
```
Comment (cptartur) [inline lib/src/main/kotlin/starknet/crypto/NativeLoader.kt:3]: I'd prefer we not use io.File at all. This is an old api. `nio.file` is newer and provides a better apis.

**Review by cptartur — CHANGES_REQUESTED** [review]

**Comment by bartekryba** [comment]
I got
`Error: Call requires API level 26 (current min is 24): java.nio.file.Files#createTempDirectory [NewApi]`
when building this

File: lib/src/main/kotlin/starknet/crypto/NativeLoader.kt (hunk 1 above)
Comment (Solpatium) [reply lib/src/main/kotlin/starknet/crypto/NativeLoader.kt:27]: Why do I need to create the file?

File: lib/src/main/kotlin/starknet/crypto/NativeLoader.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            val resource = sharedLibExtensions.map {
+                val fileName = "/lib$name.$it"
+                object {}::class.java.getResource(fileName)
+            }.firstOrNull { it != null } ?: throw e
+
+            // Android lint complains about it, but android should never reach this code
+            val tmpDir = Files.createTempDirectory("$name-dir").toFile().apply {
+                deleteOnExit()
+            }
+            val tmpFilePath = FileSystems.getDefault().getPath(tmpDir.absolutePath, name)
```
Comment (cptartur) [inline lib/src/main/kotlin/starknet/crypto/NativeLoader.kt:26]:
```suggestion
            val tmpFilePath = Paths.get(tmpDir.absolutePath, name)
```
It should work the same but is simpler. You should verify though

**Review by cptartur — COMMENTED** [review]
I think we should create a issues for:
* Improve NativeLoader and get rid of lint ignores
* Add unit tests for NativeLoader

**Review by bartekryba — APPROVED** [review]

File: lib/src/main/kotlin/starknet/crypto/NativeLoader.kt (hunk 4 above)
Comment (Solpatium) [reply lib/src/main/kotlin/starknet/crypto/NativeLoader.kt:26]: No, it doesn't work on android.

**Review by cptartur — APPROVED** [review]

**Pull request merged**